app/views/assignments/_file_manager.html.erb

```diff
@@ -6,7 +6,9 @@
           assignment_id: <%= @assignment.id %>,
           repo_url: '<%= File.join(MarkusConfigurator.markus_config_repository_external_base_url,
                                    Assignment::STARTER_CODE_REPO_NAME) %>',
-          readOnly: <%= @assignment.can_upload_starter_code? ? 'false' : 'true' %>
+          groupsExist: <%= @assignment.groups.size == 0 ? 'false' : 'true' %>,
+          authenticity_token: '<%= form_authenticity_token(form_options: { action: update_starter_code_assignment_url,
```

Comment:
The problem is caused by a discrepancy between the Rails path helpers and the `js-routes` path helpers. Looks like by default, the former uses the current locale (`en`), but the js-routes version doesn't. I guess this normally doesn't matter, but here it does because these are the strings used to generate the authenticity tokens.

You can fix this here by passing `locale: nil` to the path helper. I guess a longer-term solution would be to make `js-routes` use the current locale when generate paths. For some reason I thought it already did that, but I guess now.

Reply:
@david-yz-liu Good catch. I have implemented the `locale: nil` solution for now and made the Issue #3725 so that we can make the long-term change later on.